server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -586,6 +589,10 @@ private LookupResult lookup(SortedKeyValueIterator<Key,Value> mmfi, List<Range>
           mmfi.next();
         }
 
+      } catch (ScanYieldException sye) {
+        log.debug("Scan yield exception detected at position " + sye.getPosition());
+        addUnfinishedRange(lookupResult, range, sye.getPosition(), false);
```

Comment:
Each key that is actually returned by the iterator will be returned to the user.  The ScanYieldException would be thrown within the next call (presumably because it has been doing a lot of scanning and not actually finding any keys to return).  The key placed in the exception would where the next call left off before finding the next key to return.  All keys already returned by getTopKey() after successfull next calls would still be returned appropriately.
If you follow the timesUp or the exceededMemoryUsage used here, you will see that it is essentially follows the same paradigm of yielding the scan until later.  I am just supplying a way for the underlying iterator to force this to happen.